docs/source/checks/nlp/model_evaluation/plot_metadata_segments_performance.py

```diff
@@ -0,0 +1,107 @@
+   features (out of the ones selected above) to predict the per sample error calculated before.
+
+#. We convert each of the leafs in each of the trees into a segment and calculate the segment's performance. For the
+   weakest segments detected we also calculate the model's performance on data segments surrounding them.
+"""
+#%%
+# Generate data & model
+# =====================
+
+from deepchecks.nlp.datasets.classification.tweet_emotion import load_data, load_precalculated_predictions
+
+_, test_dataset = load_data(data_format='TextData')
+_, test_probas = load_precalculated_predictions(pred_format='probabilities')
+
+test_dataset.metadata.head(3)
+
+#%%
+# Run the check
+# =============
+#
+# The check has several key parameters (that are all optional) that affect the behavior of the
+# check and especially its output.
+#
+# ``columns / ignore_columns``: Controls which columns should be searched for weak segments. By default,
+# uses all columns.
+#
+# ``alternative_scorer``: Determines the metric to be used as the performance measurement of the model on different
+# segments. It is important to select a metric that is relevant to the data domain and task you are performing.
+#
+# ``segment_minimum_size_ratio``: Determines the minimum size of segments that are of interest. The check is tuned
+# to find the weakest segment regardless of the segment size and so it is recommended to try different configurations
+# of this parameter as larger segments can be of interest even the model performance on them is superior.
+#
+# ``categorical_aggregation_threshold``: By default the check will combine rare categories into a single category called
+# "Other". This parameter determines the frequency threshold for categories to be mapped into to the "other" category.
+#
+# see :class:`API reference <deepchecks.tabular.checks.model_evaluation.WeakSegmentsPerformance>` for more details.
+
+from deepchecks.nlp.checks import MetadataSegmentsPerformance
+from sklearn.metrics import make_scorer, f1_score
+
+scorer = {'f1': make_scorer(f1_score, average='micro')}
+check = MetadataSegmentsPerformance(alternative_scorer=scorer,
+                                    segment_minimum_size_ratio=0.03)
+result = check.run(test_dataset, probabilities=test_probas)
+result.show()
+
+#%%
+# Observe the check's output
+# --------------------------
+#
+# We see in the results that the check indeed found several segments on which the model performance is below average.
+# In the heatmap display we can see model performance on the weakest segments and their environment with respect to the
+# two features that are relevant to the segment. In order to get the full list of weak segments found we will inspect
+# the result.value attribute.
+
+
+result.value['weak_segments_list'].head(3)
+
+#%%
+# Define a condition
+# ==================
+#
+# We can define on our check a condition that will validate that the model performance on the weakest segment detected
```

Comment:
We can add a condition that will validate the model's performance on the ...

Reply:
I think also worth explaining in a sentence why this is a wanted condition (why this is straightforward, I'm not sure it will be clear when reading the current sentence, for someone that isn't familiar with this check)